client-index/src/auto_sync_core.rs

```diff
@@ -236,29 +254,16 @@ where
         match id {
             // this is special, it's command
             "add_wallet" => {
-                let name = value["wallet"]["name"].as_str().chain(|| {
-                    (
-                        ErrorKind::DeserializationError,
-                        format!(
-                            "Unable to deserialize `wallet.name` from RPC data: {}",
-                            value
-                        ),
-                    )
-                })?;
+                let info: AddWalletCommand =
+                    serde_json::from_value(value).expect("get AddWalletCommand");
```

Comment:
I guess this could use the error reporting `.chain` instead of a panic @devashishdxt  ?

Reply:
yes, ".chain" is better
i'll modify